energy-integration/fees-collector/src/fees_accumulation.rs

```diff
@@ -59,6 +55,37 @@ pub trait FeesAccumulationModule:
         }
     }
 
+    fn try_swap_to_base_token(&self, payment: &mut EsdtTokenPayment) {
```

Comment:
I don't think this works as intended. We want to give all rewards in XMEX. This means we need to either go further by swapping the WEGLD to MEX and accumulate that. Or the other option would be, whenever we claim the rewards for a week, the first claim operation in the week does an extra swap of all the gathered WEGLD to MEX and then we distribuite the total MEX token.